File: app/src/main/java/com/moh/mhealth/PatientDao.java
package com.moh.mhealth;

import java.util.List;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

@Dao
public interface PatientDao {

    @Insert
    public void insert(Patient patient);

    @Update
    public void updatePatient(Patient patient);

    @Query("SELECT * FROM patients")
    public List<Patient> getAllPatients();

    @Query("SELECT pid, name, dob FROM patients")
    public List<PatientListTuple> loadPatients();

    @Query("SELECT pid, name, dob FROM patients WHERE name LIKE :name")
    public List<PatientListTuple> loadPatients(String name);

    @Query("SELECT * FROM patients WHERE pid = :pid")
    public Patient getPatient(int pid);

}

File: app/src/main/java/com/moh/mhealth/PatientDao.kt
package com.moh.mhealth

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface PatientDao {
    @Insert
    suspend fun insert(patient: Patient)

    @Update
    suspend fun updatePatient(patient: Patient)

    @Query("SELECT * FROM patients")
    fun getAllPatients(): Flow<List<Patient>>

    @Query("SELECT pid, name, dob FROM patients")
    fun loadPatients(): Flow<List<PatientListTuple>>

    @Query("SELECT pid, name, dob FROM patients WHERE name LIKE :name")
    suspend fun loadPatients(name: String): List<PatientListTuple>

    @Query("SELECT * FROM patients WHERE pid = :pid")
    suspend fun getPatient(pid: Int): List<Patient>
}